cloudformation/loadbalancer.yml

```diff
@@ -0,0 +1,172 @@
+      Runtime: nodejs6.10
+      Code:
+        ZipFile: !Sub |
+          const aws = require('aws-sdk');
+          const cloudformation = new aws.CloudFormation();
+          const response = require('cfn-response');
+
+          function customResourceTemplate(handler) {
+            return function(event, context) {
+              console.log("REQUEST RECEIVED:\n" + JSON.stringify(event));
+
+              const resolve = response.send.bind(this, event, context, response.SUCCESS);
+              const reject = response.send.bind(this, event, context, response.FAILED);
+              if(event.RequestType === 'Delete') {
+                // Nothing to do to delete the resource
+                resolve()
+                return;
+              }
+
+              try {
+                handler(event).then(resolve).catch(reason => {
+                  console.log('HANDLER ERROR:\n' + reason);
+                  reject(reason);
+                });
+              } catch(reason) {
+                console.log('HANDLER ERROR:\n' + reason);
+                reject(reason);
+              }
+            }
+          }
+          
+          // Return a mapping of Logical Id to Physical Id for all resources of a specific stack
+          exports.stackResources = customResourceTemplate(event => {
+            const StackName = event.ResourceProperties.StackName;
+            if(typeof StackName !== 'string')
+              throw new Error('StackName property required');
+            return cloudformation.describeStackResources({ StackName }).promise()
+              .then(result => result.StackResources.reduce((prev, cur) => {
+                prev[cur.LogicalResourceId] = cur.PhysicalResourceId;
+                return prev;
+              }, {}));
+          });
+  # Invoke the custom resource
+  ClusterStackResources:
+    Type: Custom::StackResources
+    Properties:
+      ServiceToken: !GetAtt StackResourcesLambda.Arn
+      StackName: !Join
+        - ""
+        -
+          - "EC2ContainerService-"
+          - !Ref ClusterName
+  # Create the load balancer
+  LoadBalancerSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: "Allow ELB access from internet"
+      VpcId: !GetAtt ClusterStackResources.Vpc
+      SecurityGroupIngress:
+        - IpProtocol: "tcp"
+          FromPort: "80"
+          ToPort: "80"
+          CidrIp: "0.0.0.0/0"
+        - IpProtocol: "tcp"
+          FromPort: "443"
+          ToPort: "443"
+          CidrIp: "0.0.0.0/0"
+  ExternalLoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Scheme: "internet-facing"
+      Subnets:
+        - !GetAtt ClusterStackResources.PubSubnetAz1
+        - !GetAtt ClusterStackResources.PubSubnetAz2
+      LoadBalancerAttributes:
+        - Key: idle_timeout.timeout_seconds
+          Value: "50"
+      SecurityGroups:
+        - !GetAtt LoadBalancerSecurityGroup.GroupId
+  ExternalTargetGroup:
```

Comment:
I see the direction here is to export the target group then use it for the application.
This however ties the LB to a single application.
What about having this TG as a default/dummy one, export the http/https listeners then have each application create it's own target group and add rules to listeners based on the host header.

This way one could use the same cluster/LB for multiple applications if he wants to and if he wants a dedicated LB for a app, he just creates a new LB stack and connects only one app to it.